zerver/lib/widget.py

```diff
@@ -76,3 +79,95 @@ def do_widget_post_save_actions(message: MutableMapping[str, Any]) -> None:
+        parser.blockprocessors.register(markdown.blockprocessors.HRProcessor(parser), 'hr', 70)
+        parser.blockprocessors.register(bugdown.ListIndentProcessor(parser), 'indent', 60)
+        return parser
+
+    def build_inlinepatterns(self) -> markdown.util.Registry:
+
+        NOT_STRONG_RE = markdown.inlinepatterns.NOT_STRONG_RE
+        DEL_RE = r'(?<!~)(\~\~)([^~\n]+?)(\~\~)(?!~)'
+        EMPHASIS_RE = r'(\*)(?!\s+)([^\*^\n]+)(?<!\s)\*'
+        ENTITY_RE = markdown.inlinepatterns.ENTITY_RE
+        STRONG_EM_RE = r'(\*\*\*)(?!\s+)([^\*^\n]+)(?<!\s)\*\*\*'
+        BACKTICK_RE = r'(?:(?<!\\)((?:\\{2})+)(?=`+)|(?<!\\)(`+)(.+?)(?<!`)\3(?!`))'
+
+        reg = markdown.util.Registry()
+        reg.register(markdown.inlinepatterns.DoubleTagPattern(STRONG_EM_RE, 'strong,em'), 'strong_em', 100)
+        reg.register(bugdown.ModalLink(r'!modal_link\((?P<relative_url>[^)]*), (?P<text>[^)]*)\)'), 'modal_link', 75)
+        reg.register(bugdown.LinkInlineProcessor(markdown.inlinepatterns.LINK_RE, self), 'link', 60)
+        reg.register(bugdown.AutoLink(get_web_link_regex(), self), 'autolink', 55)
+        reg = self.register_realm_filters(reg)
+        reg.register(markdown.inlinepatterns.HtmlInlineProcessor(ENTITY_RE, self), 'entity', 40)
+        reg.register(markdown.inlinepatterns.SimpleTextInlineProcessor(NOT_STRONG_RE), 'not_strong', 20)
+        return reg
+
+    def register_realm_filters(self, inlinePatterns: markdown.util.Registry) -> markdown.util.Registry:
+            for (pattern, format_string, id) in self.getConfig("realm_filters"):
+                inlinePatterns.register(RealmFilterPattern(pattern, format_string, self),
+                                        'realm_filters/%s' % (pattern,), 45)
+            return inlinePatterns
+
+    def build_treeprocessors():
+        treeprocessors = markdown.util.Registry()
+        treeprocessors.register(markdown.treeprocessors.InlineProcessor(self), 'inline', 25)# think its needed
+        treeprocessors.register(markdown.treeprocessors.PrettifyTreeprocessor(self), 'prettify', 20)
+        treeprocessors.register(bugdown.InlineInterestingLinkProcessor(self), 'inline_interesting_links', 15)# think its needed
+        if settings.CAMO_URI:
+            treeprocessors.register(bugdown.InlineHttpsProcessor(self), 'rewrite_to_https', 10)
+        return treeprocessors
+
+    def build_postprocessors():
+
+        postprocessors = markdown.util.Registry()
+        postprocessors.register(markdown.postprocessors.RawHtmlPostprocessor(self), 'raw_html', 20)
+        postprocessors.register(markdown.postprocessors.AndSubstitutePostprocessor(), 'amp_substitute', 15)
+        postprocessors.register(markdown.postprocessors.UnescapePostprocessor(), 'unescape', 10)
+        return postprocessors
+
+    def handle_zephyr_mirror():
+        if self.getConfig("realm") == ZEPHYR_MIRROR_BUGDOWN_KEY:
+            self.inlinePatterns = get_sub_registry(self.inlinePatterns, ['autolink'])
```

Comment:
These lines here indicate you haven't actually ran this code yet; this function isn't available inside this class and you'd have have to change it to `bugdown.get_sub_registry`.